plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2155,6 +2356,7 @@ public void finishMerge(ConnectorSession session, ConnectorMergeTableHandle merg
             appendAddFileEntries(transactionLogWriter, newFiles, partitionColumns, getExactColumnNames(handle.getMetadataEntry()), true);
 
             transactionLogWriter.flush();
+            cacheTableMetadataAsync(handle.getSchemaName(), handle.getTableName(), Optional.ofNullable(handle.getMetadataEntry().getDescription()), commitVersion, handle.getMetadataEntry().getSchemaString());
```

Comment:
Was the caching of the metadata in the method `writeTransactionLogForInsertOperation()` left out on purpose?

Reply:
The caching logic already exists in finishInsert method. I don't feel much benefit to move to the nested method.